### ISSUE TITLE
Bug in fromDataURL for high DPI screen devices

### ISSUE THREAD

**Issue by siggifv** [szimek/signature_pad#153]
# Description

I've found that the method `fromDataURL` behaves oddly on devices that don't have the pixel ratio 1, e.g. on mobile devices with high DPI.
I'm aware of the [clause in the README](https://github.com/szimek/signature_pad/blob/master/README.md#handling-high-dpi-screens) where handling of high DPI screens is discussed. 
However, I'm still able to reproduce an issue using `fromDataURL` in cases where the canvas is of a fixed size. I believe that [this scaling of the image](https://github.com/szimek/signature_pad/blob/master/signature_pad.js#L124-L125) is the reason for this issue.
You can reproduce this issue using [this JSFiddle](https://jsfiddle.net/siggifv/1oLghs5w/).
## Steps to reproduce
1. Open the JSFiddled linked in the description above on a device that **does not have a pixel ratio of 1** and write something on the signature pad.
2. Click Save.
   ![1_after_save](https://cloud.githubusercontent.com/assets/1277384/13671375/85dcdadc-e6c7-11e5-932d-195f76b09a3b.PNG)
3. Click Edit.
4. Write some more on the pad.
   ![2_before_cancel](https://cloud.githubusercontent.com/assets/1277384/13671378/8f9fe186-e6c7-11e5-884f-7d4bb2997182.PNG)
5. Click Cancel
### Expected outcome

The signature pad should look exactly like it did after step 2.
### Actual outcome

The changes on the signature pad are reverted, but the image has been scaled so it looks different from what was on the pad after step 2.
![3_after_cancel](https://cloud.githubusercontent.com/assets/1277384/13671384/9407cd24-e6c7-11e5-94de-1925664bc8fa.PNG)

**Comment by durasj** [comment]
Hi,

I also stumbled upon this issue. This is unexpected behavior. I think we should have a choice whether we want such modifications (so second argument defaulting to previous behavior should be fine). Mainly because something like that is not done automatically anywhere else. 

Thanks a lot @siggifv for your investigation, you saved me some time.

Manually populating canvas is the way for now:

``` js
                    var image = new Image();
                    signaturePad.clear();
                    image.src = dataURL;
                    image.onload = function () {
                        signatureCanvas = document.querySelector("#signature-pad");
                        signatureCanvas.getContext("2d").drawImage(image, 0, 0, width, height);
                    };
                    this.signaturePad._isEmpty = false;
```

**Comment by RonB** [comment]
Hi Durasj,
thanks a lot... have been struggling and used your solution in my angular directive.

**Comment by djleonskennedy** [comment]
Hi @durasj Thanks a lot, work perfectly

my example with accessor's `writeValue`

```
  writeValue(value: string): void {
    if (!value) {
      this.signaturePad.clear();
      return;
    }
   this.value = value;
   this._drawValueToCanvas();
   this.cdr.markForCheck();
  }

  /** Draw from image according to retina or some devices  */
  _drawValueToCanvas() {
    if (this.canvas != null) {
      const image = new Image();
      this.signaturePad.clear();
      image.src = this.value;
      image.onload = () =>
        this.canvas
          .getContext('2d')
          .drawImage(image, 0, 0, 300, 120);
      this.signaturePad._isEmpty = false;
    }
  }
```

**Comment by UziTech** [comment]
fixed in #253
